fe/fe-core/src/main/java/com/starrocks/binlog/BinlogManager.java

```diff
@@ -0,0 +1,266 @@
+
+                allTableIds.remove(tableId);
+                binlogTxnIdToTableId.remove(binlogTxnId);
+            }
+        }
+        transactionidToBinlogTxnId.remove(toPublishTxnId);
+        binlogLock.readLock().lock();
+    }
+
+    // the caller don't need to hold the db lock
+    public boolean tryEnableBinlog(Database db, long tableId, long binlogTtL, long binlogMaxSize) {
+        // after the FE metadata is successfully modified,
+        // flag of context will be set to true
+        BinlogContext context = new BinlogContext();
+        // pass properties not binlogConfig is for
+        // unify the logic of alter table manaul and alter table of mv trigger
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_ENABLE, "true");
+        if (!(binlogTtL == INVALID)) {
+            properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_TTL, String.valueOf(binlogTtL));
+        }
+        if (!(binlogMaxSize == INVALID)) {
+            properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_MAX_SIZE, String.valueOf(binlogMaxSize));
+        }
+
+        try {
+            SchemaChangeHandler schemaChangeHandler = GlobalStateMgr.getCurrentState().getSchemaChangeHandler();
+            schemaChangeHandler.updateBinlogConfigMeta(db, tableId, properties, context, TTabletMetaType.BINLOG_CONFIG);
+        } catch (DdlException exception) {
+            return context.isModifyFeMetaSuccess();
+        }
+        return true;
+
+    }
+
+    // the caller don't need to hold the db lock
+    public boolean tryCloseBinlog(Database db, long tableId) {
+        try {
+            // modify FE meta, return true
+            // try to distrucite BE task best effort
+            HashMap<String, String> properties = new HashMap<>();
+            properties.put(PropertyAnalyzer.PROPERTIES_BINLOG_ENABLE, "false");
+            SchemaChangeHandler schemaChangeHandler = GlobalStateMgr.getCurrentState().getSchemaChangeHandler();
+            schemaChangeHandler.updateBinlogConfigMeta(db, tableId, properties, null, TTabletMetaType.DISABLE_BINLOG);
+        } catch (DdlException e) {
+            // in fact, no exception will be thrown when disable binlog,
+            // it is just sharing a method，updateTableMeta，
+            // which may throw exception when enable binlog
+            LOG.warn(e.getMessage());
+            return false;
+        }
+        return true;
+    }
+
+    // Check whether the binlogAvailableVersion of all partitions of the table is available,
+    // return true if available
+    // return false if anyone is unavailable,
+    public boolean isBinlogAvailable(long tableId, long dbId) {
+        OlapTable olaptable = (OlapTable) GlobalStateMgr.getCurrentState().getDb(dbId).getTable(tableId);
+        long failedNum = olaptable.getPartitions().stream().filter(partition -> {
+            return partition.getBinlogAvailableVersion() == -1;
```

Comment:
What if the partition is created dynamically? The default available version for the new partition is alway `-1` and will not be updated.